### PR TITLE
Feature + Fix: Chocfactorykeybinds

### PR DESCRIPTION
**Pull request by catgirlseraid**
## Dependencies
- #1641 

## What
Adds keybinds to buy hoppity upgrades. This uses the same code as the keybinds for the harp so i presume it's allowed?
Fixed harp keybinds still allowing vanilla keybinds for swapping slots through

## Changelog New Features
+ Added keybinds for Chocolate Factory. - seraid

## Changelog Fixes
+ Fixed harp keybinds. - seraid

### CODE REVIEW

**Comment by catgirlseraid** [comment]
there is a edge case where two items can be brought with one keypress if hovering over an item and pressing a keybind for a different item. this happens as the vanilla keypress is not canceled

**Comment by hannibal002** [comment]
> there is a edge case where two items can be brought with one keypress if hovering over an item and pressing a keybind for a different item. this happens as the vanilla keypress is not canceled

The same edge case exists in harp keybind. If you find a good fix for this, please be so kind and also fix it in harp keybind

File: src/main/java/at/hannibal2/skyhanni/config/features/inventory/chocolatefactory/ChocolateFactoryKeybindsConfig.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+public class ChocolateFactoryKeybindsConfig {
+    @Expose
+    @ConfigOption(name = "Keybinds On", desc = "Turns the keybinds on or off")
+    @ConfigEditorBoolean
+    @FeatureToggle
+    public boolean chocolateFactoryKeybinds = false;
+
+    @Expose
+    @ConfigOption(name = "Key 1", desc = "Key for Rabbit Bro")
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/config/features/inventory/chocolatefactory/ChocolateFactoryKeybindsConfig.java]: dot at the end of the desc (and below)

File: src/main/java/at/hannibal2/skyhanni/config/features/inventory/chocolatefactory/ChocolateFactoryKeybindsConfig.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import at.hannibal2.skyhanni.config.FeatureToggle;
+import com.google.gson.annotations.Expose;
+import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorBoolean;
+import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorKeybind;
+import io.github.notenoughupdates.moulconfig.annotations.ConfigOption;
+import org.lwjgl.input.Keyboard;
+
+public class ChocolateFactoryKeybindsConfig {
+    @Expose
+    @ConfigOption(name = "Keybinds On", desc = "Turns the keybinds on or off")
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/config/features/inventory/chocolatefactory/ChocolateFactoryKeybindsConfig.java]: better desc

File: src/main/java/at/hannibal2/skyhanni/features/inventory/chocolatefactory/ChocolateFactoryKeybinds.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.milliseconds
+
+object ChocolateFactoryKeybinds {
+    private val config get() = ChocolateFactoryAPI.config.chocolateFactoryKeybindsConfig
+    private var lastClick = SimpleTimeMark.farPast()
+
+    private object KeyIterable : Iterable<Int> {
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/inventory/chocolatefactory/ChocolateFactoryKeybinds.kt]: overly complicated. simply use `0..4` instead

**Review by hannibal002 — CHANGES_REQUESTED** [review]
not tested in game yet

File: src/main/java/at/hannibal2/skyhanni/features/inventory/chocolatefactory/ChocolateFactoryKeybinds.kt (hunk 3 above)
Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/features/inventory/chocolatefactory/ChocolateFactoryKeybinds.kt]: fixed

**Pull request merged**